File: 240. Search a 2D Matrix II/main.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int R = matrix.size();
        if(R == 0)
            return false; 
        int C = matrix[0].size(); 
        int I = 0, J = C-1;
        while(I<R && J>=0){
            if(matrix[I][J] == target)
                return true; 
            if(matrix[I][J] < target)
                I++;
            else if(matrix[I][J] > target)
                J--; 
        }
        return false; 
    }
};

File: 240. Search a 2D Matrix II/main.py
class Solution(object):
    def searchMatrix(self, matrix, target):
        """
        :type matrix: List[List[int]]
        :type target: int
        :rtype: bool
        """
        
        m = len(matrix)
        if m==0:
            return False
        n = len(matrix[0])
        
        r = 0
        c = n-1
        
        while r<m and c>=0:
            if matrix[r][c] == target:
                return True
            elif matrix[r][c] < target:
                r += 1
            elif matrix[r][c] > target:    
                c -= 1
                
        return False
